### PR TITLE
[doc] Documentation update for --show-suppressed flag

### LINKED ISSUE #5229: [doc]  CLI flag `--show-suppressed`  needs to mention xml, html, summaryhtml
**Affects PMD Version:** 

Documentation

**Description:**

On the CLI flags, it does not say that only xml and html report formats will show suppressed violations, but that's what the source code says. Update the documentation to specify only those two outputs.

(Am submitting a PR which I will reference here.)

**Exception Stacktrace:**

```
# Copy-paste the stack trace here
```

**Code Sample demonstrating the issue:**

--show-suppressed has no effect on CSV, for example.

**Steps to reproduce:**

Please provide detailed steps for how we can reproduce the bug.

1.  Run a scan with CSV format and `--show-suppressed` and they are not shown
2. Run with html output and the flag, and it shows suppressed.

**Running PMD through:** *CLI*

### PR DESCRIPTION
**Pull request by dschach**
## Describe the PR

On the CLI flags, it does not say that only xml and html report formats will show suppressed violations, but that's what the source code says. Updates the documentation to specify only those two outputs.

## Related issues

#5229 

- Fixes #5229

## Ready?

- [ ] Added unit tests for fixed bug/feature
- [ ] Passing all unit tests
- [ ] Complete build `./mvnw clean verify` passes (checked automatically by github actions)
- [ ] Added (in-code) documentation (if needed)

### CODE REVIEW

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">No regression tested rules have been changed.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: docs/pages/pmd/userdocs/cli_reference.md [hunk 1]
```diff
@@ -159,7 +159,8 @@ The tool comes with a rather extensive help text, simply running with `--help`!
                description="Path to a file to which report output is written. The file is created if it does not exist. If this option is not specified, the report is rendered to standard output."
     %}
     {% include custom/cli_option_row.html options="--show-suppressed"
-               description="Causes the suppressed rule violations to be added to the report."
+               description="Causes the suppressed rule violations to be added to the report (xml and html formats only)."
+               languages="html, xml"
```
Comment (adangel) [inline docs/pages/pmd/userdocs/cli_reference.md:163]: This options is not dependent on the language, but on the renderer, that is used (option `--format`). So we shouldn't add the "languages" attribute here.

I guess, this other sentence should be updated as well:
https://github.com/pmd/pmd/blob/1acaedc158e733cdd10ca771be97417c55991e37/docs/pages/pmd/userdocs/pmd_report_formats.md?plain=1#L21

something like: "... and if the format supports showing suppressed violations. Currently only html, summaryhtml and xml show suppressed violations."

We actually have this already here:
https://github.com/pmd/pmd/blob/1acaedc158e733cdd10ca771be97417c55991e37/docs/pages/pmd/userdocs/pmd_report_formats.md?plain=1#L15

> Not all report formats display all information.

Both are from this documentation page: https://docs.pmd-code.org/latest/pmd_userdocs_report_formats.html

Note: for the CSV format, we have this explicitly documented:
https://docs.pmd-code.org/latest/pmd_userdocs_report_formats.html#csv

> Suppressed violations or errors are ignored.

Comment (adangel) [inline docs/pages/pmd/userdocs/cli_reference.md:163]:
```suggestion
               description="Causes the suppressed rule violations to be added to the report if supported by the report format. See [PMD Report formats](pmd_userdocs_report_formats.html) for details."
```

If we change this here, we should also change it in the code, so that this information is also shown at the command line (`--help`):
https://github.com/pmd/pmd/blob/1acaedc158e733cdd10ca771be97417c55991e37/pmd-cli/src/main/java/net/sourceforge/pmd/cli/commands/internal/PmdCommand.java#L141

```diff
-@Option(names = "--show-suppressed", description = "Report should show suppressed rule violations.")
+@Option(names = "--show-suppressed", description = "Report should show suppressed rule violations if supported by the report format.")
```

**Review by adangel — CHANGES_REQUESTED** [review]
Thanks for your PR!

Could you have a look at my comments? I think, we should improve the doc at some more places as well.

**Comment by dschach** [comment]
Had some trouble with branch renaming. Thanks for updating to main from master.
I updated docs in the files you indicated.

File: docs/pages/pmd/userdocs/pmd_report_formats.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 title: Report formats for PMD
 tags: [pmd, userdocs]
 keywords: [formats, renderers]
-summary: "Overview of the built-in report formats for PMD"
+summary: 'Overview of the built-in report formats for PMD'
```
Comment (adangel) [inline docs/pages/pmd/userdocs/pmd_report_formats.md:5]: I'd prefer to not do unrelated changes...

File: docs/pages/pmd/userdocs/pmd_report_formats.md [hunk 3]
```diff
@@ -49,10 +49,10 @@ the full rule details for each reported rule violation.
 
 Example:
 
-```
+````
```
Comment (adangel) [inline docs/pages/pmd/userdocs/pmd_report_formats.md:52]: Is this intentional?

**Review by adangel — APPROVED** [review]
Thanks for the update.

> Had some trouble with branch renaming.

I see, you created the PR off your master branch - please create in the future a feature branch, before you create a PR. This PR has now also the commits from your other PR (#5202) - although they seem different commits.
I'm going to cleanup/rebase this PR and force-push to your master branch...

File: docs/pages/pmd/userdocs/pmd_report_formats.md (hunk 3 above)
Comment (dschach) [reply docs/pages/pmd/userdocs/pmd_report_formats.md:52]: I checked a few times, and Prettier kept doing that. No clue why; I assumed it knew something I don't. Happy to edit the files

File: docs/pages/pmd/userdocs/pmd_report_formats.md (hunk 2 above)
Comment (dschach) [reply docs/pages/pmd/userdocs/pmd_report_formats.md:5]: Also Prettier formatting. Next time I'll disable it
